gprofiler/profilers/factory.py

```diff
@@ -24,44 +28,67 @@ def get_profilers(
     process_profilers_instances: List["ProcessProfilerBase"] = []
     system_profiler: Union["SystemProfiler", "NoopProfiler"] = NoopProfiler()
 
-    if profiling_mode != "none":
-        arch = get_arch()
-        for profiler_name, profiler_config in get_profilers_registry().items():
-            lower_profiler_name = profiler_name.lower()
-            profiler_mode = user_args.get(f"{lower_profiler_name}_mode")
-            if profiler_mode in ("none", "disabled"):
-                continue
-
-            supported_archs = (
-                profiler_config.supported_windows_archs if is_windows() else profiler_config.supported_archs
-            )
-            if arch not in supported_archs:
+    if profiling_mode == "none":
+        return system_profiler, process_profilers_instances
+    arch = get_arch()
+    for runtime in get_profilers_registry():
+        runtime_args_prefix = runtime.lower().replace("-", "_")
+        runtime_mode = user_args.get(f"{runtime_args_prefix}_mode")
```

Comment:
1. Why do you need `replace(-, _)` on the runtime name?
2. This logic (these 2 lines) repeat more than once, extract to a function?